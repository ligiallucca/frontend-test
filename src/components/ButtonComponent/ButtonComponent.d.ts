import React from "react";
import { ButtonProps } from "./ButtonComponent.types";
declare const ButtonComponent: React.FC<ButtonProps>;
export default ButtonComponent;
//# sourceMappingURL=ButtonComponent.d.ts.map