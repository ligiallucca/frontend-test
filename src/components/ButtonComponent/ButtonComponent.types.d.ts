export interface ButtonProps {
    text: string;
    type?: "button" | "submit" | "reset";
    disabled?: boolean;
    onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
}
//# sourceMappingURL=ButtonComponent.types.d.ts.map