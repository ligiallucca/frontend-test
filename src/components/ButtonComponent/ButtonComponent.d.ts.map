{"version":3,"file":"ButtonComponent.d.ts","sourceRoot":"","sources":["ButtonComponent.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAC;AAE1B,OAAO,EAAE,WAAW,EAAE,MAAM,yBAAyB,CAAC;AA2BtD,QAAA,MAAM,eAAe,EAAE,KAAK,CAAC,EAAE,CAAC,WAAW,CAgB1C,CAAC;AAEF,eAAe,eAAe,CAAC"}