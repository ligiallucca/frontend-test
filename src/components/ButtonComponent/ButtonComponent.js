import { jsx as _jsx } from "react/jsx-runtime";
import styled from "styled-components";
const StyledButton = styled.button `
  width: 100%;
  border: none;
  outline: none;
  cursor: pointer;
  font-weight: 700;
  font-size: 1.2rem;
  border-radius: 4px;
  padding: 14px 20px;
  letter-spacing: 1px;
  font-family: "Inter", sans-serif;
  transition: background-color 0.3s ease-in-out;
  color: ${({ theme }) => theme.colors.white};
  background-color: ${({ theme }) => theme.colors.primary};

  &:hover {
    background-color: ${({ theme }) => theme.colors.darkenPrimary};
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;
const ButtonComponent = ({ text, onClick, disabled = false, type = "button", }) => {
    return (_jsx(StyledButton, { id: text.toLowerCase(), onClick: onClick, disabled: disabled, type: type, children: text }));
};
export default ButtonComponent;
