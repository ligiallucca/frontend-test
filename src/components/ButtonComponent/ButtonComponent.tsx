import React from "react";
import styled from "styled-components";
import { ButtonProps } from "./ButtonComponent.types";

const StyledButton = styled.button`
  width: 100%;
  border: none;
  outline: none;
  font-size: 1.5rem;
  font-weight: 700;
  border-radius: 4px;
  padding: 14px 20px;
  letter-spacing: 1px;
  font-family: "Inter", sans-serif;
  transition: background-color 0.3s ease-in-out;
  color: ${({ theme }) => theme.colors.white};
  background-color: ${({ theme }) => theme.colors.primary};

  &:hover {
    background-color: ${({ theme }) => theme.colors.darkenPrimary};
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ButtonComponent: React.FC<ButtonProps> = ({
  text,
  onClick,
  disabled,
}) => {
  return (
    <StyledButton id={text.toLowerCase()} onClick={onClick} disabled={disabled}>
      {text}
    </StyledButton>
  );
};

export default ButtonComponent;
