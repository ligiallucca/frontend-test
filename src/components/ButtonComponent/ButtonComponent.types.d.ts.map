{"version":3,"file":"ButtonComponent.types.d.ts","sourceRoot":"","sources":["ButtonComponent.types.ts"],"names":[],"mappings":"AAAA,MAAM,WAAW,WAAW;IAC1B,IAAI,EAAE,MAAM,CAAC;IACb,IAAI,CAAC,EAAE,QAAQ,GAAG,QAAQ,GAAG,OAAO,CAAC;IACrC,QAAQ,CAAC,EAAE,OAAO,CAAC;IACnB,OAAO,CAAC,EAAE,CAAC,CAAC,EAAE,KAAK,CAAC,UAAU,CAAC,iBAAiB,CAAC,KAAK,IAAI,CAAC;CAC5D"}