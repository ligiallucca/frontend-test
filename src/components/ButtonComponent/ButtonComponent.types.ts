export interface ButtonProps {
  text: string;
  disabled?: boolean;
  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
}
