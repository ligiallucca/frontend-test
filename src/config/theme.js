export const theme = {
    colors: {
        primary: "#EC6724",
        darkenPrimary: "#E15813",
        gray: "#D9D9D9",
        darkenGray: "#AAAAAA",
        ligthGray: "#E9ECF1",
        white: "#FFFFFF",
        black: "#000000",
        blackShadow: "#0000001A",
        red: "#E60000",
    },
};
