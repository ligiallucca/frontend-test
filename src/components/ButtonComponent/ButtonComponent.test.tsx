import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import ButtonComponent from "./ButtonComponent";
import { ThemeProvider } from "styled-components";

import { vi } from "vitest";
import { theme } from "../../config/theme";

const renderButtonComponent = () => {
  render(
    <ThemeProvider theme={theme}>
      <ButtonComponent text="Texto" onClick={() => {}} />
    </ThemeProvider>
  );
};

const renderButtonDisabledComponent = () => {
  render(
    <ThemeProvider theme={theme}>
      <ButtonComponent text="Texto" onClick={() => {}} disabled={true} />
    </ThemeProvider>
  );
};

test("renders button with text", () => {
  renderButtonComponent();
  const buttonElement = screen.getByText(/texto/i);
  expect(buttonElement).toBeInTheDocument();
});

test("button is disabled when disabled prop is true", () => {
  renderButtonDisabledComponent();
  const buttonElement = screen.getByText(/texto/i);
  expect(buttonElement).toBeDisabled();
});

test("calls onClick function when button is clicked", () => {
  const handleClick = vi.fn();
  renderButtonComponent();
  const buttonElement = screen.getByText(/texto/i);
  fireEvent.click(buttonElement);
  waitFor(() => expect(handleClick).toHaveBeenCalled());
});

test("button has correct background color", () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <ButtonComponent text="Texto" onClick={() => {}} />
    </ThemeProvider>
  );
  const buttonElement = container.querySelector("button");

  expect(buttonElement).not.toBeNull();

  if (buttonElement)
    expect(buttonElement).toHaveStyle(
      `background-color: ${theme.colors.darkenPrimary}`
    );
});

test("button changes background color on hover", () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <ButtonComponent text="Texto" onClick={() => {}} />
    </ThemeProvider>
  );
  const buttonElement = container.querySelector("button");
  if (buttonElement) fireEvent.mouseOver(buttonElement);
  expect(buttonElement).toHaveStyle(
    `background-color: ${theme.colors.darkenPrimary}`
  );
});

test("button does not have outline on focus", () => {
  renderButtonComponent();
  const buttonElement = screen.getByText(/Texto/i);
  buttonElement.focus();
  expect(buttonElement).toHaveStyle("box-shadow: none");
});

test("matches snapshot", () => {
  const { asFragment } = render(
    <ThemeProvider theme={theme}>
      <ButtonComponent text="Texto" onClick={() => {}} />
    </ThemeProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
