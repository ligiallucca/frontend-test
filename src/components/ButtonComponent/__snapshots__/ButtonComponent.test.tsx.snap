// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-blHHSb bbmPQL"
    id="texto"
  >
    Texto
  </button>
</DocumentFragment>
`;
